pkg/connector/tickets.go

```diff
@@ -54,6 +109,109 @@ func getJiraStatusesForProject(ctx context.Context, client *jira.Client, project
 	return jiraStatuses, nil
 }
 
+func (j *Jira) constructMetaDataFields(issues []*jira.MetaIssueType) (map[string]jira.MetaDataFields, error) {
+	fieldsMap := make(map[string]jira.MetaDataFields)
+
+	for _, issueType := range issues {
+		for key, field := range issueType.Fields {
+			var metaDataField jira.MetaDataFields
+
+			jsonData, err := json.Marshal(field)
+			if err != nil {
+				return nil, err
+			}
+
+			err = json.Unmarshal(jsonData, &metaDataField)
+			if err != nil {
+				return nil, err
+			}
+
+			fieldsMap[key] = metaDataField
+		}
+	}
+
+	return fieldsMap, nil
+}
+
+func (j *Jira) getCustomFieldsForProject(ctx context.Context, projectKey string, issueTypeIDs []string) ([]*v2.TicketCustomField, error) {
+	metadata, _, err := j.client.Issue.GetCreateMeta(ctx, &jira.GetQueryOptions{
+		ProjectKeys:  projectKey,
+		Expand:       "projects.issuetypes.fields",
+		IssueTypeIds: strings.Join(issueTypeIDs, ","),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	j.metaProject = metadata.Projects[0]
+
+	project := metadata.Projects[0] // we should only be getting one project back
+	fieldsMap, err := j.constructMetaDataFields(project.IssueTypes)
+	if err != nil {
+		return nil, err
+	}
+
+	customFields := make([]*v2.TicketCustomField, 0, len(fieldsMap))
+	for _, field := range fieldsMap {
+		var customField *v2.TicketCustomField
+		var allowedValues []*v2.TicketCustomFieldObjectValue
+
+		if !field.Required || field.Schema.Custom == "" {
+			continue
+		}
+
+		hasAllowedValues := len(field.AllowedValues) > 0
+		isMultiSelect := field.Schema.Items != ""
+
+		if hasAllowedValues {
+			for _, choice := range field.AllowedValues {
+				displayName := choice.Name
+				if displayName == "" {
+					displayName = choice.Value
+				}
+				allowedValues = append(allowedValues, &v2.TicketCustomFieldObjectValue{
+					Id:          choice.Id,
+					DisplayName: displayName,
+				})
+			}
+		}
+
+		id := field.Key
+
+		switch field.Schema.Type {
+		case jira.TypeString:
+			customField = sdkTicket.StringFieldSchema(id, field.Name, false)
+		case jira.TypeArray:
+			switch {
+			case isMultiSelect && hasAllowedValues:
+				customField = sdkTicket.PickMultipleObjectValuesFieldSchema(id, field.Name, false, allowedValues)
+			case !isMultiSelect && hasAllowedValues:
+				customField = sdkTicket.PickObjectValueFieldSchema(id, field.Name, false, allowedValues)
+			case isMultiSelect && !hasAllowedValues:
```

Comment:
nit: I think it would be helpful to have this after `case isMultiSelect && hasAllowedValues:` for readability